src/gleam/bool.gleam

```diff
@@ -323,3 +323,36 @@ pub fn to_string(bool: Bool) -> String {
     True -> "True"
   }
 }
+
+/// Run a callback function if the given bool is `True`, otherwise return a
+/// default value.
+///
+/// This function is suitable for `use` expressions.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > let name = "Kamaka"
+/// > use <- guard(name != "", or: "friend")
+/// > "Hello, " <> name
+/// "Hello, Kamaka"
+/// ```
+///
+/// ```gleam
+/// > let name = ""
+/// > use <- guard(name != "", or: "friend")
+/// > "Hello, " <> name
+/// "Hello, friend"
+/// ```
+///
+///
+pub fn guard(
+  requirement: Bool,
+  or alternative: t,
```

Comment:
This has the unfortunate consequence of making it very easy to drop in an expensive computation as the alternative, run it, and then never use it.

It's a caveat that often comes up with these sorts of things (including `with_default`/`unwrap`): it's definitely caught me out in Elm code before.